### ISSUE TITLE
Improve UI for Dashboard Sidebar with Many Components

### ISSUE THREAD

**Issue by K009** [highcharts/highcharts#21977]
# Description of the feature
In the current component sidebar (Edit Mode > Add Components), users are presented with a long, scrollable list when selecting from a large number of available components. It would be good to enhance the usability of the component sidebar for example by: 
- Implementing groupings of related components, allowing users to expand or collapse sections. This would significantly reduce the need to scroll through long lists and provide more structured navigation.
- Adding a search bar or filter system so that users can quickly locate specific components by typing keywords or selecting predefined categories (e.g., frequently used, recently added, by type).
- Allowing users to resize the component sidebar pane so they can view more components without needing to scroll as much.

# Library related to the feature
Highcharts Dashboards

# Proof of Concept/Live example for the feature
![image](https://github.com/user-attachments/assets/1e0fe614-4194-46ee-bd9a-457a3fb48af7)

----

**You can vote for this feature by adding a thumbs-up reaction to this post.**

**Comment by redherring917b** [comment]
Agreed that options for dealing with a component sidebar (editMode > Add Components) - format, layout, expandable / shrinkable groupings, etc. - when dealing with a large number of components would be very much welcomed.  Just offering the user one long, scrollable list of all available components doesn't seem like the best UI option.

Thanks!
